### PR TITLE
Improve Normalize Whitespace for block lambdas

### LINKED ISSUE #46656: Improve NormalizeWhitespace for multi block lambdas
**Version Used**:
I'm linking:

```xml
		<PackageReference Include="Microsoft.CodeAnalysis.CSharp.Workspaces" Version="3.7.0-4.final" PrivateAssets="all" />
		<PackageReference Include="Microsoft.CodeAnalysis.Analyzers" Version="3.3.0-beta1.final" PrivateAssets="all" />
```

**Steps to Reproduce**:

Call NormalizeWhitespace() on the following csharp code:
```csharp
class C
{
     void M()
     {
           M1(() => {});
     }
     void M1(Action a){}
}
```

**Expected Behavior**:
Normalizes to:

```csharp
class C
{
     void M()
     {
           M1(() => 
           {
           });
     }

     void M1(Action a)
     {
     }

}
```

**Actual Behavior**:

```csharp
class C
{
     void M()
     {
           M1(() =>
           {
           }

           );
     }

     void M1(Action a)
     {
     }

}
```

I'm happy to fix this myself if this is the sort of breaking change you're prepared to take.

### PR DESCRIPTION
**Pull request by YairHalberstadt**
Fixes #46656

### CODE REVIEW

File: src/Compilers/CSharp/Portable/Syntax/SyntaxNormalizer.cs [hunk 1]
```diff
@@ -282,7 +282,8 @@ private static int LineBreaksAfterOpenBrace(SyntaxToken currentToken, SyntaxToke
         private static int LineBreaksAfterCloseBrace(SyntaxToken currentToken, SyntaxToken nextToken)
         {
             if (currentToken.Parent is InitializerExpressionSyntax ||
-                currentToken.Parent.IsKind(SyntaxKind.Interpolation))
+                currentToken.Parent.IsKind(SyntaxKind.Interpolation) ||
+                currentToken.Parent?.Parent is ParenthesizedLambdaExpressionSyntax or SimpleLambdaExpressionSyntax)
```
Comment (CyrusNajmabadi) [inline src/Compilers/CSharp/Portable/Syntax/SyntaxNormalizer.cs:286]: What about anonymous methods?
Also, can just do `is LambdaExpression`

File: src/Compilers/CSharp/Test/Syntax/Syntax/SyntaxNormalizerTests.cs [hunk 2]
```diff
@@ -565,6 +565,24 @@ public void TestNormalizeTuples()
...
+            TestNormalizeStatement("Add(()=>{{_=x=>{};}});",
+@"Add(() =>
+{
+  {
+    _ = x =>
+    {
+    };
+  }
+});");
+        }
```
Comment (CyrusNajmabadi) [inline src/Compilers/CSharp/Test/Syntax/Syntax/SyntaxNormalizerTests.cs:584]: Would prefer a theory here. That way if one fails it's trivial to just run that config

**Comment by 333fred** [comment]
@dotnet/roslyn-compiler for a second review.

**Comment by jcouv** [comment]
Merged/squashed. Thanks @YairHalberstadt!

**Pull request merged**